client/webserver/site/src/js/app.ts

```diff
@@ -549,15 +559,37 @@ export default class Application {
         else if (!updateOrder(mkt, order)) mkt.orders.push(order)
         break
       }
+      case 'fiatrateupdate': {
+        const r: RateNote = note as RateNote
+        this.fiatRatesMap = r.fiatRates
+        if (!this.walletMap) break
+        for (const [assetID, wallet] of (Object.entries(this.walletMap) as [any, WalletState][])) {
+          if (!wallet) continue
+          const rate = r.fiatRates[assetID]
+          const fiatDiv = this.main.querySelector(`[data-conversion-target="${assetID}"]`) as PageElement
+          if (!fiatDiv || !fiatDiv.firstElementChild) continue
+          fiatDiv.firstElementChild.textContent = Doc.formatAssetValue(wallet.balance.available, rate, this.unitInfo(assetID))
+          if (rate && !this.disableConversion) Doc.show(fiatDiv)
+          else Doc.hide(fiatDiv)
+        }
+      }
+        break
       case 'balance': {
         const n: BalanceNote = note as BalanceNote
         const asset = this.user.assets[n.assetID]
         // Balance updates can come before the user is fetched after login.
         if (!asset) break
         const w = asset.wallet
-        if (w) w.balance = n.balance
-        break
+        if (!w) break
+        w.balance = n.balance
+        const rate = this.fiatRatesMap[w.assetID]
+        const fiatDiv = this.main.querySelector(`[data-conversion-target="${n.assetID}"]`) as PageElement
+        if (!fiatDiv || !fiatDiv.firstElementChild) return
+        fiatDiv.firstElementChild.textContent = Doc.formatAssetValue(n.balance.available, rate, this.unitInfo(n.assetID))
+        if (rate && !this.disableConversion) Doc.show(fiatDiv)
+        else Doc.hide(fiatDiv)
```

Comment:
This 5-line block belongs in `wallets.ts`, since that's the only place that has the `data-conversion-target` elements. You already have it there, but you can delete it from **app.ts** in two places.